examples/common/offworld_wrapper.py

```diff
@@ -0,0 +1,173 @@
+        if self.channel == 1: #depth only
+            depth = frame[0, :, :, -1]
+            depth = (np.clip(depth / 5, 0, 1) * 255).astype('uint8')  # convert depth to uint8
+            obs = depth
+        elif self.channel == 3: #rgb only:
+            rgb = cv2.cvtColor(frame[0, :, :, :3].astype('float32'), cv2.COLOR_RGB2GRAY).astype('uint8')
+            obs = rgb
+        else: # rgbd
+            depth = frame[0, :, :, -1]
+            depth = (np.clip(depth / 5, 0, 1) * 255).astype('uint8')  # convert depth to uint8
+            rgb = cv2.cvtColor(frame[0, :, :, :3].astype('float32'), cv2.COLOR_RGB2GRAY).astype('uint8')
+            obs = np.stack([rgb, depth], axis=-1)
+
+        obs_resized = cv2.resize(obs, (self.size, self.size), interpolation=cv2.INTER_AREA)
+        obs_resized = obs_resized[..., np.newaxis] if len(obs_resized.shape) == 2 else obs_resized
+        return np.transpose(obs_resized, axes=[2, 0, 1])
+
+class FrameStack(gym.Wrapper):
+    """Stack n_frames last frames.
+
+    :param gym.Env env: the environment to wrap.
+    :param int n_frames: the number of frames to stack.
+    """
+
+    def __init__(self, env, n_frames):
+        super().__init__(env)
+        self.n_frames = n_frames
+        self.frames = deque([], maxlen=n_frames)
+        shape = (env.observation_space.shape[0] * n_frames, *env.observation_space.shape[1:])
+        self.observation_space = gym.spaces.Box(
+            low=np.min(env.observation_space.low),
+            high=np.max(env.observation_space.high),
+            shape=shape,
+            dtype=env.observation_space.dtype
+        )
+
+    def reset(self):
+        obs = self.env.reset()
+        for _ in range(self.n_frames):
+            self.frames.append(obs)
+        return self._get_ob()
+
+    def step(self, action):
+        obs, reward, done, info = self.env.step(action)
+        self.frames.append(obs)
+        return self._get_ob(), reward, done, info
+
+    def _get_ob(self):
+        # the original wrapper use `LazyFrames` but since we use np buffer,
+        # it has no effect
+        return np.concatenate(self.frames, axis=0)
+
+def wrap_offworld(
+    env_id,
+    frame_stack=1,
+    warp_frame=True,
+):
+    """Configure environment for DeepMind-style Atari. The observation is
+    channel-first: (c, h, w) instead of (h, w, c).
+
+    :param str env_id: the atari environment id.
+    :param bool episode_life: wrap the episode life wrapper.
+    :param bool clip_rewards: wrap the reward clipping wrapper.
+    :param int frame_stack: wrap the frame stacking wrapper.
+    :param bool scale: wrap the scaling observation wrapper.
+    :param bool warp_frame: wrap the grayscale + resize observation wrapper.
+    :return: the wrapped atari environment.
+    """
+    env = gym.make(env_id,channel_type=Channels.RGBD)
+    if warp_frame:
+        env = WarpFrame(env)
+    if frame_stack:
+        env = FrameStack(env, frame_stack)
+    return env
+
+def wrap_offworld_real(
```

Comment:
This function and the one above look almost identical. Please merge them into one by passing `channel_type` as argument (for example using `**kwargs`)

Reply:
Yes, just the arguments of env = gym.make()  are diffferent for sim and real, I can move env = gym.make() outside of wrapper, and pass the env instance as input to the wrapper.

Reply:
This is a good solution!